portfolio/src/main/java/com/google/sps/servlets/UsersServlet.java

```diff
@@ -0,0 +1,50 @@
+
+package com.example.appengine.users;
+
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+// With @WebServlet annotation the webapp/WEB-INF/web.xml is no longer required.
+@WebServlet(
+    name = "UserAPI",
+    description = "UserAPI: Login / Logout with UserService",
+    urlPatterns = "/userapi"
+)
+public class UsersServlet extends HttpServlet {
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+
+    String thisUrl = req.getRequestURI();
+
+    resp.setContentType("text/html");
+    if (req.getUserPrincipal() != null) {
+      resp.getWriter()
+          .println(
+              "<p>Hello, "
+                  + req.getUserPrincipal().getName()
+                  + "!  You can <a href=\""
+                  + userService.createLogoutURL(thisUrl)
+                  + "\">sign out</a>.</p>"
+                  + "</br>"
+                  +"<p> <a href=\"index.html\">Back to home page </a></p>");
+                  //TODO: add in link to go back to homepage 
+                  // add in nickname field
+                  
+    } else {
+      resp.getWriter()
+          .println(
+              "<p>Please <a href=\"" 
+              + userService.createLoginURL(thisUrl) 
+              + "\">sign in</a>.</p>"
+              + "</br>"
+              +"<p> <a href=\"index.html\">Back to home page </a></p> ");
+    }
+  }
+}
```

Comment:
Looks like there's a warning about no newline at the of the file, please take a look.